Core/Core/Features/Files/View/FileList/FileListViewController.swift

```diff
@@ -116,6 +127,14 @@ public class FileListViewController: ScreenViewTrackableViewController, ColoredN
             self.tableView.contentOffset.y = self.searchBar.frame.height
         }
 
+        studentAccessInteractor?
+            .isRestricted()
+            .receive(on: DispatchQueue.main)
+            .sink { [weak self] isRestricted in
+                self?.isRestricted = isRestricted
+            }
+            .store(in: &subscriptions)
```

Comment:
In rare cases it could happen that the cache invalidates before moving to this page so the `isRestricted()` method makes an API call that takes time to complete. The screen can load during this time for the user with `isRestricted` being `false`, allowing the user to upload a file.
A better approach would be to keep the loading view on screen until we receive a response from the `isRestricted()` method call. Another solution would be to make the `isRestricted()` call to be synchronous by using a cached value but that would require to load the state of this flag at app start.